### PR TITLE
[Jellyfin] Old browsers

### PR DESCRIPTION
**Pull request by dmitrylyzo**
This PR improves compatibility with older browsers.
Jellyfin is focused on supporting webOS 1.2 and Tizen 2.3, which use ancient web engines (WebKit).

I cherry-picked related commits, skipping merge-commits.

<details>
<summary>List of commits</summary>

82822b25340c2c3bd557710bf9e5ac7ef39af8de
64445220b42d1111eaafd7279afe857904efdef9
9299074ee500ad1c14ee8daac4fac7511524c9c2
`Merge pull request #15 from JustAMan/update-build`
other commits are new
</details>

### CODE REVIEW

**Review by TFSThiagoBR98 — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@
 BASE_DIR:=$(dir $(realpath $(firstword $(MAKEFILE_LIST))))
 DIST_DIR:=$(BASE_DIR)dist/libraries
 
-GLOBAL_CFLAGS:=-O3
+GLOBAL_CFLAGS:=-O3 -s ENVIRONMENT=web,webview -s DOUBLE_MODE=0
```
Comment (TFSThiagoBR98) [inline Makefile:7]: DOUBLE_MODE has been removed from Emscripten, see [here](https://github.com/emscripten-core/emscripten/blob/aab919ddfe92e88c8f1d67159d41e2291af7a1dd/src/settings.js#L2034), @dmitrylyzo can you remove it here?

**Review by TheOneric — COMMENTED** [review]
Thanks for your PRs! Assuming you spot-checled this PR doesn't break anything with JSO's example suit (the `gh_pages` branch); just a few nits:

Normally I'd like to see fixups done by amending the faulty/outdated patch-commit rather than adding yet another commit; howver since this is porting from JustAMan's jellyfin patches there's some value in keeping the commits as unchanged as possible. **But** your last two `Remove unnecessary flags (already in GLOBAL_CFLAGS) ` and `Remove deprecated flag (PR review) ` commits should imho be squashed into one `Drop superfluous and deprecated flags` commit.
**Also** it would be nice if you could add `Cherry-pciked from: https://github.com/jellyfin/JavascriptSubtitlesOctopus/commit/<commit-hash>` to the description of the cherry-picked commits.

---
If you haven't yet checked the examples: In order to test your changes, checkout the `gh_pages` branch in a seperate folder; update the `assets/js`folder to what building your PR yields and run a local server. Eg create a symlink named `JavascriptSubtitlesOctopus` in and to the root folder of the gh_pages repo and run this Python script: [run-server.py](https://github.com/libass/JavascriptSubtitlesOctopus/files/7239539/run-server.py.txt). Your tests will be available at `http://localhost:8000/JavscriptSubtitlesOctopus`


You'll want to at least play the `Video.JS` sample until `01:10` and skip a bit through the `Railgun S Karaoke Test`, `Attack on Titan Stress Karaoke Test` and the `Brotli Compressed Subtitle File` examples and make sure the subtitles are rendered and not obviously borked.

**Comment by dmitrylyzo** [comment]
Done. Also, simplified `from` polyfills a bit.

Tested `Video.JS`, `Railgun S Karaoke Test`, `Attack on Titan Stress Karaoke Test` and `Brotli Compressed Subtitle File` - no issues except these errors in the console (the same for the JSO `master`):
jso-err1
Fontconfig error occurs when cache is disabled.

All these polyfills should probably be fixed in emscripten itself (https://github.com/emscripten-core/emscripten/issues/11008 and https://github.com/emscripten-core/emscripten/pull/11011)

**Comment by TheOneric** [comment]
Thanks again!
Looks good afaict and since Thiago also already approved, I'll merge it shortly.

> no issues except these errors in the console (the same for the JSO master)

Yeah, the MIMe and following “ArrayBuffer instatiation” can be safely ignored, they are only because the test server setup doesn't know about the wasm-MIME type (and emscripten will fall back to fetching it anyway via “ArrayBuffer instatiation”) and well we don't have a favicon either. Not sure about  the Fontconfig error, but it's not a regression and iinm will degrade font-lookup performance, but not break anything.

> All these polyfills should probably be fixed in emscripten itself (emscripten-core/emscripten#11008 and emscripten-core/emscripten#11011)

Having emscripten provide the necessary polyfills would indeed be ideal.

**Pull request merged**